proposals/0000-integrations.md

```diff
@@ -0,0 +1,171 @@
+- The `hooks: {}` API conceptually matches Rollup & Vite but was designed to avoid the risk of conflict that would have been introduced had we literally extending the Vite plugin idea. 
+	- This is why we prefix all hooks with `astro:`.
+
+
+# Drawbacks
+
+- **Breaking changes across an integration system are expensive.** This can be mitigated until v1.0.0, see adoption strategy below.
+
+
+# Alternatives
+
+A previous design used a functional API more like this:
+
+```
+export default function(integration) {
+	integration.injectScript(...);
+	integration.modifyConfig(...);
+	integration.configureServer(...);
+	integration.mountDirectory(...);
+}
+```
+
+This produced nice, linear code but at the expense of internal Astro complexity and limited flexibility that eventually blocked some integrations from being possible:
+
+1. **Complexity:** Astro had to run the entire integration upfront on startup, and then cache these results for later when needed. Many of the methods (like `configureServer`) ended up acting more like hooks anyway.
+2. **Inflexible:** Integrations like Partytown couldn't work with a provided `mountDirectory` helper method because because they need to run their own `fs` logic on the final build directory. 
+
+Advanced use-cases like this essentially required hooks to perform custom logic as needed, so the design shifted away from "helpers that do everything for you" and towards "provide hooks with helpers availble if needed".
+
+# Adoption strategy
+
+## Experimental Flag
+
+This proposal suggests only supporting official integrations to start, and mark 3rd-part integrations as experimental via something like a config flag (`--experimental-integrations`) until we hit `v1.0.0-beta`.
+
+This would let us test the integration system and respond to user feedback before finalizing.
+
+## Renderers
+
+The `renderers` API is deprecated by this proposal, with all `renderers` becoming `integrations`: `@astrojs/renderer-vue` -> `@astrojs/vue`.
+
+With a lot of work, we could do this in a backwards compatible way. However, I would like to avoid that complexity (and the potential for bugs that comes with it) and do this in a breaking change for the following reasons:
+
+1. **low-effort to upgrade:** updating your renderers to integrations would involve changing your config file only. A codemod could be provided to make this even easier.
+1. **easy to assist:** Unlike past breaking changes, this will be fairly easy for Astro to provide helpful output to migrate from one to the other.
+
+```
+$ astro build
+
+Astro renderers are no longer supported!
+Update your config to use Astros new integration system:
+
+- renderers: ["@astrojs/vue"]
++ integrations: [import("@astrojs/vue")]
+```
+
+# Unresolved questions
+
+- Bikeshedding all the things
+- Can we pair this with an `astro add NAME` CLI command?
```

Comment:
cc @juanM04 who started exploring this in https://github.com/withastro/astro/pull/2650